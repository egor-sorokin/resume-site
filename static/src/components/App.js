import React, {Component} from 'react';
import PersonalData from './PersonalData';
import WorkExperience from './WorkExperience';

export default class App extends Component {
  render() {
    return (
      <div>
        <PersonalData />
        <WorkExperience />
      </div>
    );
  }
}
