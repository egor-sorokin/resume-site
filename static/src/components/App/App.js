import React, {Component} from 'react';
import PersonalData from './../PersonalData';
import WorkExperience from './../WorkExperience';
import Educations from './../Educations/Educations';
import Skills from './../Skills';
import Projects from './../Projects';

export default class App extends Component {
  render() {
    return (
      <div>
        <PersonalData />
        <WorkExperience />
        <Educations />
        <Projects />
        <Skills />
      </div>
    );
  }
}
