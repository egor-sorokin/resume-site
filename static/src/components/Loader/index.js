import React from 'react';
import './styles.scss';

const Loader = () => (
  <div className="loader">
    <h3 className="loader__inner font-s-20 text-c-l-dune">
      Loading...
    </h3>
  </div>
);

export default Loader;
