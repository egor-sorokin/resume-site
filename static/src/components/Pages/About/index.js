import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {TimelineMax} from 'gsap';
import {attachToggleAnimation} from './animation';
import OrderedList from '../../OrderedList/index'
import Summary from '../../Summary/index'
import Logo from '../../Logo/index';
import LinkStretched from '../../LinkStretched/index';
import {LINK_MASKED, BUTTON_CLOSE} from '../../../constants/index';
import './styles.scss';


class About extends Component {
  constructor(props) {
    super(props);

    this.aboutTween = null;
  }


  componentDidMount() {
    if (!this.props.isFetching) {
      this.aboutTween = new TimelineMax();
      attachToggleAnimation(this.aboutTween);
    }
  }


  componentDidUpdate(prepProps) {
    if (this.props.isOpenedAbout !== prepProps.isOpenedAbout) {
      this.aboutTween.reversed(!this.props.isOpenedAbout);
      document.body.style.overflowY = (this.props.isOpenedAbout ? 'hidden' : 'initial');
    }
  }


  clickCloseButton = () => {
    this.props.toggleAboutSection();
  };


  render() {
    const {data} = this.props;
    const personData = data.personData || {};
    const socials = personData.socials || [];
    const cssClassesButton = "button-close__link font-s-12-secondary text-c-mercury-light";

    return (
      <div>
        <section className="about text-c-mercury-light">
          <div className="about__item about__item--left">
            <div className="line line--top"></div>
            <Logo></Logo>
            <div className="line line--bottom"></div>
          </div>
          <div className="about__item about__item--middle">
            <h1 className="about__title font-s-36"
            >
              {personData.first_name + ' ' + personData.last_name}
            </h1>
            <Summary
              summary={personData.summary}
            ></Summary>
          </div>
          <div className="about__item about__item--right">
            <OrderedList
              title={socials.title}
              items={socials.items}
              type={LINK_MASKED}
            ></OrderedList>
          </div>
          <div className="button-close">
            <LinkStretched
              onClick={this.clickCloseButton}
              cssClasses={cssClassesButton}
              text={BUTTON_CLOSE.text}
            ></LinkStretched>
          </div>
        </section>
      </div>
    );
  }
}


About.propTypes = {
  isFetching: PropTypes.bool,
  data: PropTypes.shape({}),
  isOpenedAbout: PropTypes.bool,
  toggleAboutSection: PropTypes.func
};

About.defaultProps = {
  data: {},
  isFetching: true,
  isOpenedAbout: false,
  toggleAboutSection: () => {
  }
};


export default About;
