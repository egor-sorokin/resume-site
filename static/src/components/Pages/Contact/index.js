import React from 'react';
import PropTypes from 'prop-types';
import Logo from '../../Logo/index';
import UnorderedList from '../../UnorderedList/index';
import Navbar from '../../Navbar/index';
import withFetching from '../../../utils/api';
import {URL_PATH_PERSON_DATA} from '../../../constants/index';
import './styles.scss';


const Contact = ({data, isFetching, error}) => {
  const personData = data.personData || {};
  const socials = personData.socials || [];
  const navbarItems = ['Home', 'About', 'Projects', 'Experiments'];

  if (error) {
    return <div><p>{error}</p></div>;
  }

  if (isFetching) {
    return <div><p>Loading...</p></div>;
  }

  return (
    <div>
      <section className="banner contact">
        <div className="contact-inner">
          <div className="contact-inner__item">
            <Navbar
              items={navbarItems}
            ></Navbar>
            <div className="delimiter"></div>
          </div>
          <div className="contact-inner__item">
            <Logo></Logo>
          </div>
          <div className="contact-inner__item">
            <UnorderedList
              title=''
              items={socials.items}
            ></UnorderedList>
          </div>
        </div>
      </section>
    </div>
  );
};

Contact.propTypes = {
  data: PropTypes.shape({}),
  isFetching: PropTypes.bool,
  error: PropTypes.string,
};

Contact.defaultProps = {
  data: {},
  error: '',
  isFetching: false,
};


export default withFetching(URL_PATH_PERSON_DATA, Contact);
