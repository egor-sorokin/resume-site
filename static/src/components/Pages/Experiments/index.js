import React from 'react';
import PropTypes from 'prop-types';

import InnerSlide from '../../InnerSlide';


const Experiments = ({ experiments }) => (
  <main id="experiments">
    <InnerSlide
      key="experiments"
      name={experiments.name}
      info={experiments.info}
      showcases={experiments.showcases}
    />
  </main>
);


Experiments.propTypes = {
  experiments: PropTypes.shape({
    name: PropTypes.string,
    info: PropTypes.string,
    showcases: PropTypes.instanceOf(Array),
  }),
};

Experiments.defaultProps = {
  experiments: {},
};


export default Experiments;
