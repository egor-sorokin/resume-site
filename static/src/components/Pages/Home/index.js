import React, {Component} from 'react';
import PropTypes from 'prop-types';
import scrollToY from 'scroll-to-y';
import ButtonExplore from './ButtonExplore';
import LinkStretched from '../../LinkStretched/index';
import withFetching from '../../../utils/api';
import {URL_PATH_PERSON_DATA, BUTTON_EXPLORE, BUTTON_ABOUT} from '../../../constants/index';
import './styles.scss';


class Home extends Component {
  scrollToProjects = (section, scrollTo) => {
    if (scrollTo) {
      scrollToY(
        document.getElementById(section).offsetTop,
        500,
        'easeInOutQuint'
      );
    }
  };


  clickAboutButton = () => {
    this.props.toggleAboutSection();
  };


  render() {
    const {data, isFetching, error} = this.props;
    const personData = data.personData || {};

    if (error) {
      return <div><p>{error}</p></div>;
    }

    if (isFetching) {
      return <div><p>Loading...</p></div>;
    }

    return (
      <div>
        <section id="home" className="home banner text-c-mercury-light">
          <h3 className="home__name font-s-36">
            {personData.first_name + ' ' + personData.last_name}
          </h3>

          <ButtonExplore
            onClick={this.scrollToProjects}
          >{BUTTON_EXPLORE.text}</ButtonExplore>

          <div className="switcher switcher--white">
            <div className="switcher__line line--top"/>
            <LinkStretched
              onClick={this.clickAboutButton}
              cssClasses="switcher__link font-s-12-secondary text-c-dune"
              text={BUTTON_ABOUT.text}
            ></LinkStretched>
            <div className="switcher__line line--bottom"/>
          </div>
        </section>
      </div>
    );
  }
}

Home.propTypes = {
  data: PropTypes.shape({}),
  isFetching: PropTypes.bool,
  error: PropTypes.string,
  toggleAboutSection: PropTypes.func
};

Home.defaultProps = {
  data: {},
  isFetching: false,
  error: '',
  toggleAboutSection: () => {
  }
};


export default withFetching(URL_PATH_PERSON_DATA, Home);
