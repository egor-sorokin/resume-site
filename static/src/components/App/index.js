import React, {Component} from 'react';
import Home from '../Pages/Home/index';
import Contact from '../Pages/Contact/index';
import About from '../Pages/About/index';
import Projects from '../Pages/Projects/index';
import Loader from '../Loader/index';
import RequestError from '../RequestError/index';
import fetchData from '../../utils/api';
import {URL_PATH_PERSON_DATA, URL_PATH_PROJECTS} from '../../constants/index';


class App extends Component {
  constructor() {
    super();

    this.state = {
      personData: {},
      projectsData: {},
      isOpenedAbout: false
    }
  }


  componentDidMount() {
    this._fetchData();
  };


  _fetchData() {
    this.setState({isFetching: true});

    const promisePersonData = fetchData(URL_PATH_PERSON_DATA);
    const promiseProjectsData = fetchData(URL_PATH_PROJECTS);

    Promise.all([promisePersonData, promiseProjectsData])
      .then((results) => {
        this.setState({
          personData: results[0],
          projectsData: results[1]
        })
      })
      .catch(error => this.setState({error: error.message}))
      .finally(() => this.setState({isFetching: false}));
  }


  toggleAboutSection = () => {
    this.setState({isOpenedAbout: !this.state.isOpenedAbout});
  };


  render() {
    if (this.state.error) {
      return <RequestError></RequestError>;
    }

    if (this.state.isFetching) {
      return <Loader></Loader>;
    }

    return (
      <div>
        <About
          isFetching={this.state.isFetching}
          data={this.state.personData}
          isOpenedAbout={this.state.isOpenedAbout}
          toggleAboutSection={this.toggleAboutSection}
        />
        <Home
          isFetching={this.state.isFetching}
          data={this.state.personData}
          toggleAboutSection={this.toggleAboutSection}
        />
        <Projects
          data={this.state.projectsData}
        />
        <Contact
          data={this.state.personData}
          toggleAboutSection={this.toggleAboutSection}
        />
      </div>
    )
  }
}

export default App;
