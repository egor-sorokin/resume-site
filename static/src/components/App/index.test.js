// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import App from './index';
import {API_DELAY} from '../../constants/index';
import Home from '../Pages/Home/index';
import About from '../Pages/About/index';
import Contact from '../Pages/Contact/index';
import Projects from '../Pages/Projects/index';
import Loader from '../Loader/index';
import RequestError from '../RequestError/index';
import Experiments from "../Pages/Experiments";
// import RequestError from '../RequestError/index';


describe('<App />', () => {
  let component;
  const errorMessage = 'error message';
  const person = {
    id: 1,
    first_name: 'Bob',
    last_name: 'Cat',
    email: 'example@admin.com',
    summary: [],
  };
  const projects = {
    id: 1,
    name: "Southwestern University",
    url: "https://www.southwestern.edu",
    info: [
      {
        label: "role",
        text: "Web Software Engineer",
        url: ""
      }
    ],
    image: {
      src: "assets/pr_su.jpg",
      alt: "Southwestern University"
    }
  };
  const experiments = {};


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<App />);
    });


    it('matches snapshot', async() => {
      await new Promise(res => setTimeout(() => {
        expect(component).toMatchSnapshot();

        res();
      }, API_DELAY * 2))
    });


    it('renders 5 components', async() => {
      // timeout is equal to api delay + 1ms
      await new Promise(res => setTimeout(() => {
        expect(component.children()).toHaveLength(4);
        expect(component.find(Home)).toBeTruthy();
        expect(component.find(About)).toBeTruthy();
        expect(component.find(Projects)).toBeTruthy();
        expect(component.find(Experiments)).toBeTruthy();
        expect(component.find(Contact)).toBeTruthy();

        res();
      }, API_DELAY * 2))
    });
  });


  describe('states tests', () => {
    beforeEach(() => {
      component = shallow(<App />);
    });


    it('renders loading indicator', () => {
      component.setState({isFetching: true});
      expect(component.find(Loader)).toHaveLength(1);
      expect(component.state().isFetching).toBe(true);
      component.setState({isFetching: false, person: person});
    });


    it('renders error message', () => {
      component.setState({error: errorMessage});
      expect(component.find(RequestError)).toHaveLength(1);
      expect(component.state().error).toBe(errorMessage);
      component.setState({error: null});
    });


    it('renders person', () => {
      component.setState({person: {person}});
      expect(component.state().person.person).toEqual(person);
      component.setState({person: {}});
    });


    it('renders projects', () => {
      component.setState({projects: {projects}});
      expect(component.state().projects.projects).toEqual(projects);
      component.setState({projects: {}});
    });

    it('renders experiments', () => {
      component.setState({ experiments });
      expect(component.state().experiments).toEqual(experiments);
      component.setState({experiments: {}});
    });
  });


  describe('lifecycle tests', () => {
    it('calls componentDidMount', () => {
      const spy = jest.spyOn(App.prototype, 'componentDidMount');
      // eslint-disable-next-line
      component = mount(<App />);
      expect(spy).toHaveBeenCalledTimes(1);

      afterEach(() => {
        spy.mockClear();
      });
    });
  });
});
