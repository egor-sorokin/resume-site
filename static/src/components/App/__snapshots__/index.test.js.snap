// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div>
  <WithFetching />
  <WithFetching />
  <WithFetching />
  <WithFetching />
  <WithFetching />
</div>
`;
